src/fidgets/farcaster/components/CreateCast.tsx

```diff
@@ -0,0 +1,297 @@
+    return true;
+  };
+
+  const isPublishing = draft?.status === DraftStatus.publishing;
+  const {
+    editor,
+    getText,
+    addEmbed,
+    getEmbeds,
+    setEmbeds,
+    setChannel,
+    getChannel,
+    handleSubmit,
+    setText,
+  } = useEditor({
+    fetchUrlMetadata: getUrlMetadata,
+    onError,
+    onSubmit: onSubmitPost,
+    linkClassName: "text-blue-300",
+    renderChannelsSuggestionConfig: createRenderMentionsSuggestionConfig({
+      getResults: debouncedGetChannels,
+      RenderList: ChannelList,
+    }),
+    renderMentionsSuggestionConfig: createRenderMentionsSuggestionConfig({
+      getResults: debouncedGetMentions,
+      RenderList: MentionList,
+    }),
+    editorOptions: {
+      parseOptions: {
+        preserveWhitespace: "full",
+      },
+    },
+  });
+
+  useEffect(() => {
+    if (!text && draft?.text && isEmpty(draft.mentionsToFids)) {
+      editor?.commands.setContent(
+        `<p>${draft.text.replace(/\n/g, "<br>")}</p>`,
+        true,
+        {
+          preserveWhitespace: "full",
+        },
+      );
+    }
+
+    if (draft?.embeds) {
+      setInitialEmbeds(draft.embeds);
+    }
+  }, [editor]);
+
+  const text = getText();
+  const embeds = getEmbeds();
+  const channel = getChannel();
+
+  useEffect(() => {
+    if (!editor) return; // no updates before editor is initialized
+    if (isPublishing) return;
+    if (draft?.parentUrl === channel?.parent_url) return;
+
+    const newEmbeds = initialEmbeds ? [...embeds, ...initialEmbeds] : embeds;
+    setDraft({
+      ...draft,
+      text,
+      embeds: newEmbeds,
+      parentUrl: channel?.parent_url || undefined,
+    });
+  }, [text, embeds, initialEmbeds, channel, isPublishing, editor]);
+
+  const getButtonText = () => {
+    if (isLoadingSigner) return "Not signed into Farcaster";
+    if (isPublishing) return "Publishing...";
+    return "Cast";
+  };
+
+  return (
+    <div
+      className="flex flex-col items-start min-w-full w-full h-full"
+      tabIndex={-1}
+    >
+      <form onSubmit={handleSubmit} className="w-full">
+        {isPublishing ? (
+          <div className="w-full h-full min-h-[150px]">
+            <Skeleton className="px-2 py-1 w-full h-full min-h-[150px] text-foreground/80">
+              {draft.text}
+            </Skeleton>
+          </div>
+        ) : (
+          <div className="p-2 border-slate-200 rounded-lg border">
+            <EditorContent
+              editor={editor}
+              autoFocus
+              className="w-full h-full min-h-[150px] text-foreground/80"
+            />
+            <EmbedsEditor
+              embeds={[]}
+              setEmbeds={setEmbeds}
+              RichEmbed={() => <div />}
+            />
+          </div>
+        )}
+
+        <div className="flex flex-row pt-2 gap-1">
+          {!isReply && (
+            <div className="text-foreground/80">
+              {isPublishing || isLoadingSigner ? (
+                getChannel().name
```

Comment:
`const channel = getChannel();` is already defined above. Replace with `channel.name` instead of calling `getChannel()` again?